### ISSUE TITLE
Complete windowpost fails if a few sectors are missing

### ISSUE THREAD

**Issue by RobQuistNL** [filecoin-project/lotus#9026]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [ ] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [X] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
1.16.1
```

### Describe the Bug

Example:

10 storage machines
10000 sectors

each one has 1000 sectors

If one drops out, I would expect to have a 10% loss, so 1000 faults.

What happens is that 100% goes to fault. That shouldn't happen. Proving check shows that some are gone / can't be found, but the rest end up as good.

Except for the actual wdpost. All fail.

### Logging Information

```text
$ lotus-miner proving check --slow 0 | grep good | wc -l
2115
$ lotus-miner proving check --slow 0 | grep bad | wc -l
234
```

```
deadline  partitions  sectors (faults)  proven partitions
0         1           2349 (2344)       0
1         1           2349 (2346)       0
```
```


### Repo Steps

1. Run lotus wdpost

**Comment by magik6k** [comment]
1. Are you using PoSt workers?
2. Can you share lotus-miner logs around PoSt time?

**Comment by RobQuistNL** [comment]
Yes, using postworkers. After posting this issue I've continued researching what was going on and I think that what happened here is that an NFS share got botched and hung. This causes the entire wdpost to fail (I think because the process is waiting on returns from the storage system?)

The thing is that even my SSH session would hang if I were to try to list these mounted folders. CTRL+C would not help. So I doubt if this is fixable in Lotus..

After I forcefully unmounted this "broken" NFS share (by running umount -f a couple of times) and removing it from the lotus storage.json, its behaving as expected, and skipping the missing sectors, proving the ones that are around.

I've however noticed this stuff more. The scheduler seems to depend on the same system storage calls because once these NFS mounts start to hang, the scheduler freezes up too.

I also see a load of these everywhere (miner and workers);
```json
{"level":"warn","ts":"2022-07-11T13:10:50.653Z","logger":"advmgr","caller":"sector-storage/sched_worker.go:209","msg":"failed to check worker session","error":"RPC client error: sendRequest failed: Post \"http://10.224.2.68:4566/rpc/v0\": context deadline exceeded"}
{"level":"warn","ts":"2022-07-11T13:10:54.075Z","logger":"advmgr","caller":"sector-storage/stats.go:27","msg":"getting worker task types in WorkerStats","error":"RPC client error: sendRequest failed: Post \"http://10.224.2.68:4562/rpc/v0\": context deadline exceeded"}
```

When killing the miner process I see hundreds of thousands of these;

```json
{"level":"error","ts":"2022-07-11T13:58:42.763Z","logger":"advmgr","caller":"sector-storage/sched.go:412","msg":"trySched(1) req.sel.Ok error: getting worker paths:\n    github.com/filecoin-project/lotus/extern/sector-storage.(*existingSelector).Ok\n        /home/filecoinnew/lotus/extern/sector-storage/selector_existing.go:42\n  - RPC client error: sendRequest failed: Post \"http://10.224.1.20:3456/rpc/v0\": context deadline exceeded"}
```

So it looks like these queue up and never return (or return very late and hang the process).


so, TL;DR:
- Botched NFS mounts
- Lotus does indeed work correctly if NFS mounts work correctly
- Migth be another fix (implement better timeouts on storage calls / limit parallel storage calls)

**Comment by TippyFlitsUK** [comment]
Hi Rob

Many thanks for providing such a detailed solution! Closing this ticket as it is not a Lotus issue.
